my_class.cpp

```diff
@@ -1,151 +1,175 @@
-//@file: my_class.cpp
-//@author: Veselina Velkova (veselina.velkova@strypes.eu)
-//@description: This contains the implementation of the student class.
-//@version: 0.1
-//@date: 18.04.2023
-
 #include "my_class.h"
 #include <iostream>
 
+//This constructor is under comment because we do not need it for the purposes of the unit testing lesson.
+/*
 student::student()
 {
-    while(true)
-    {
-        std::cout << "What is student's number in class?" << '\n';
-        std::cin >> number;
-        if(!std::cin)
-        {
-            std::cerr << "Read invalid data for student's number in class!"<<std::endl;
-            std::cin.clear();
-            std::cin.ignore();
-        }
-        else
-        {
-            break;
-        }
-    }    
-    std::cin.ignore();
-    while(true)
-    {
-        std::cout << "What's the student's name?" << '\n';
-        getline(std::cin, name);
-        if(!std::cin)
-        {
-            std::cerr << "Read invalid data for the student's name!" << std::endl;
-            std::cin.clear();
-            std::cin.ignore();
-        }
-        else
-        {
-            break;
-        }
-    }
-    grades.resize(MIN_GRADES);
+   while(true)
+   {
+      std::cout << "What is student's number in class?" << '\n';
+      std::cin >> number;
+      if(!std::cin)
+      {
+         std::cerr << "Read invalid data for student's number in class!"<<std::endl;
+         std::cin.clear();
+         std::cin.ignore();
+      }
+      else
+      {
+         break;
+      }
+   }    
+   std::cin.ignore();
+   while(true)
+   {
+      std::cout << "What's the student's name?" << '\n';
+      getline(std::cin, name);
+      if(!std::cin || !isValidName(name))
+      {
+         std::cerr << "Read invalid data for the student's name!" << std::endl;
+         std::cin.clear();
+         std::cin.ignore();
+      }
+      else
+      {
+         break;
+      }
+   }
+   grades.reserve(MIN_GRADES);
+}
+*/
+
+student::student(const int num, const std::string& full_name)
+{
+   number = num;
+   if(isValidName(full_name))
+   {
+       name = full_name;
+   }
+   else
+   {
+      throw std::invalid_argument("Invalid string for student's name in constructor!");
+   }
+   grades.reserve(MIN_GRADES);
+}
+
+bool student::isValidGrade(const float grade)
+{
+   return (grade >= LOWEST_GRADE && grade <= HIGHEST_GRADE);
 }
 
-student::student(int num, const std::string& full_name)
+bool student::isValidName(const std::string& new_name)
 {
-    number = num;
-    name = full_name;
-    grades.resize(MIN_GRADES);
+   for(char ch : new_name)
+   {
+      if(ch != ' ' && ch != '-' && !isalpha(ch))
```

Comment:
I might be paranoid, but I would put explicit brackets around each of these, like this:
if((ch != ' ') && (ch != '-') && !isalpha(ch))